src/query/storages/fuse/src/operations/read.rs

```diff
@@ -62,27 +67,75 @@ impl FuseTable {
         }
     }
 
+    pub fn prewhere_of_push_downs(&self, push_downs: &Option<Extras>) -> Option<PrewhereInfo> {
+        if let Some(Extras { prewhere, .. }) = push_downs {
+            prewhere.clone()
+        } else {
+            None
+        }
+    }
+
     #[inline]
     pub fn do_read2(
         &self,
         ctx: Arc<dyn TableContext>,
         plan: &ReadDataSourcePlan,
         pipeline: &mut Pipeline,
     ) -> Result<()> {
-        let block_reader =
-            self.create_block_reader(&ctx, self.projection_of_push_downs(&plan.push_downs))?;
+        let projection = self.projection_of_push_downs(&plan.push_downs);
+        let prewhere_info = if let Some(prewhere) = self.prewhere_of_push_downs(&plan.push_downs) {
+            let schema = self.table_info.schema();
+            let expr_field = prewhere.filter.to_data_field(&schema)?;
```

Comment:
If the filter column is a inner column of tuple, we can't get `expr_field` from `schema`

Reply:
Something off topic but related to this: `delete from t where s:a = 1` is also not supported?

Reply:
> If the filter column is a inner column of tuple, we can't get expr_field from schema

So the solution is to generate a one-boolean-column schema manually?

Reply:
`delete from t where s:a = 1` is not supported as it has not migrated to new planner.

Reply:
We can generate a schema of inner columns by `path_indices`, like this:
https://github.com/datafuselabs/databend/blob/main/src/query/service/src/storages/storage_table_read_plan.rs#L75

Reply:
I think we can do just like this:

```rust
let expr_schema = DataSchemaRefExt::create(vec![DataField::new(
    prewhere.filter.column_name().as_str(),
    BooleanType::new_impl(),
)]);
```